pypdf/_writer.py

```diff
@@ -517,7 +575,15 @@ def insert_page(
         Returns:
             The added PageObject.
         """
-        return self._add_page(page, lambda kids, p: kids.insert(index, p))
+        assert self.flattened_pages is not None
```

Comment:
Shouldn't this raise a proper error?

Reply:
this is a mypy assert: flattened_pages is now properly initialized / updated to a list since the constructor